zebra-chain/src/orchard/tests/preallocate.rs

```diff
@@ -0,0 +1,95 @@
+        prop_assert!((smallest_disallowed_serialized.len() as u64) >= MAX_BLOCK_BYTES);
+
+        // Create largest_allowed_vec by removing one element from smallest_disallowed_vec without copying (for efficiency)
+        smallest_disallowed_vec.pop();
+        let largest_allowed_vec = smallest_disallowed_vec;
+        let largest_allowed_serialized = largest_allowed_vec.zcash_serialize_to_vec().expect("Serialization to vec must succeed");
+
+        // Check that our largest_allowed_vec contains the maximum number of <Action>
+        prop_assert!((largest_allowed_vec.len() as u64) == <Action>::max_allocation());
+        // Check that our largest_allowed_vec is small enough to fit in a Zcash block.
+        prop_assert!((largest_allowed_serialized.len() as u64) < MAX_BLOCK_BYTES);
+    }
+
+    /// Verify that...
+    /// 1. The smallest disallowed vector of `Signature<SpendAuth>`s is too large to fit in a Zcash block
+    /// 2. The largest allowed vector is small enough to fit in a legal Zcash block
+    #[test]
+    fn spend_auth_sig_max_allocation_is_correct(sig in <Signature<SpendAuth>>::arbitrary_with(())) {
+
+        let max_allocation: usize = <Signature<SpendAuth>>::max_allocation().try_into().unwrap();
+        let mut smallest_disallowed_vec = Vec::with_capacity(max_allocation + 1);
+        for _ in 0..(<Signature<SpendAuth>>::max_allocation()+1) {
+            smallest_disallowed_vec.push(sig.clone());
+        }
+        let smallest_disallowed_serialized = smallest_disallowed_vec.zcash_serialize_to_vec().expect("Serialization to vec must succeed");
+        // Check that our smallest_disallowed_vec is only one item larger than the limit
+        prop_assert!(((smallest_disallowed_vec.len() - 1) as u64) == <Signature<SpendAuth>>::max_allocation());
+        // Check that our smallest_disallowed_vec is too big to be included in a valid block
+        // Note that a serialized block always includes at least one byte for the number of transactions,
+        // so any serialized Vec<Action> at least MAX_BLOCK_BYTES long is too large to fit in a block.
+        prop_assert!((smallest_disallowed_serialized.len() as u64) >= MAX_BLOCK_BYTES);
+
+        // Create largest_allowed_vec by removing one element from smallest_disallowed_vec without copying (for efficiency)
+        smallest_disallowed_vec.pop();
+        let largest_allowed_vec = smallest_disallowed_vec;
+        let largest_allowed_serialized = largest_allowed_vec.zcash_serialize_to_vec().expect("Serialization to vec must succeed");
+
+        // Check that our largest_allowed_vec contains the maximum number of <Action>
+        prop_assert!((largest_allowed_vec.len() as u64) == <Signature<SpendAuth>>::max_allocation());
+        // Check that our largest_allowed_vec is small enough to fit in a Zcash block.
+        prop_assert!((largest_allowed_serialized.len() as u64) < MAX_BLOCK_BYTES);
+    }
+}
```

Comment:
There's no need to do tests on `Signature<SpendAuth>`, because it uses the same size as `Action`, so the tests for `Action::max_allocation` cover it.